components/brave_wallet/browser/brave_wallet_provider_impl.cc

```diff
@@ -25,9 +32,50 @@ BraveWalletProviderImpl::BraveWalletProviderImpl(
 
 BraveWalletProviderImpl::~BraveWalletProviderImpl() {}
 
+bool BraveWalletProviderImpl::OnAddEthereumChainRequest(
+    const std::string& json_payload,
+    RequestCallback callback) {
+  if (!delegate_)
+    return false;
+  base::Value payload;
+  if (!ParsePayload(json_payload, "params", &payload))
+    return false;
+  std::vector<EthereumChain> chains = ValueToEthereumChain(payload);
+  if (chains.empty()) {
+    return false;
+  }
+  DCHECK_LT(chains.size(), size_t(2));
+  // TODO(spylogsster): Add support for multiple chains;
+  auto chain = chains[0];
+  if (GetNetworkURL(prefs_, chain.chain_id).is_valid()) {
+    // OnChainAddedResult(std::move(callback), std::string());
+    // return true;
+  }
+  // By https://eips.ethereum.org/EIPS/eip-3085 only chain id is required
+  // we expect chain name and rpc urls as well at this time
+  if (chain.chain_id.empty() || chain.rpc_urls.empty() ||
+      chain.chain_name.empty()) {
+    // return false;
+  }
+  auto valueValue = brave_wallet::EthereumChainToValue(chain);
+  std::string chainJson;
+  if (!base::JSONWriter::Write(valueValue, &chainJson))
+    return false;
+  delegate_->RequestUserApproval(
+      chainJson,
+      base::BindOnce(&BraveWalletProviderImpl::OnChainAddedResult,
+                     weak_factory_.GetWeakPtr(), std::move(callback)));
+  return true;
+}
+
 void BraveWalletProviderImpl::Request(const std::string& json_payload,
                                       bool auto_retry_on_network_change,
                                       RequestCallback callback) {
+  std::string method = brave_wallet::ParseRequestMethodName(json_payload);
+  if (method == kAddEthereumChainMethod &&
+      OnAddEthereumChainRequest(json_payload, std::move(callback))) {
+    return;
+  }
```

Comment:
It seems you don't actually need to share this with other rpc requests.
Why not check your method in `BraveWalletJSHandler::Request` and have your own function in the Provider?

Reply:
done